diff --git a/CMakeLists.txt b/CMakeLists.txt
index b231a1e..8e3d6ca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,8 +30,11 @@ if(CMAKE_SYSTEM_NAME STREQUAL "iOS")
     endif()
 endif()
 
+option(VISIBILITY_PRESET_HIDDEN "set the library visibility preset to hidden" ON)
+if(VISIBILITY_PRESET_HIDDEN)
 set(CMAKE_C_VISIBILITY_PRESET hidden)
 set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+endif()
 
 if(COMMAND CMAKE_POLICY)
     cmake_policy(SET CMP0003 NEW)
