diff -Naur portaudio/src/os/unix/pa_unix_util.c portaudio_ohos/src/os/unix/pa_unix_util.c
--- portaudio/src/os/unix/pa_unix_util.c	2021-03-31 17:35:42.000000000 -0700
+++ portaudio_ohos/src/os/unix/pa_unix_util.c	2023-11-01 00:41:32.084358954 -0700
@@ -66,6 +66,8 @@
    Track memory allocations to avoid leaks.
  */
 
+#undef PTHREAD_CANCELED
+
 #if PA_TRACK_MEMORY
 static int numAllocations_ = 0;
 #endif
